cluster/shard_mapper.go

```diff
@@ -84,42 +108,49 @@ func NewRemoteMaper(nodeID, shardID uint64, stmt string, chunkSize int) *RemoteM
 }
 
 // Open connects to the remote node and starts receiving data.
-func (r *RemoteMapper) Open() error {
-	node, err := r.MetaStore.Node(r.nodeID)
+func (r *RemoteMapper) Open() (err error) {
+	defer func() {
+		if err != nil {
+			r.conn.MarkUnusable()
```

Comment:
Actually, why is every connection marked as unusable?  That should only be called when it shouldn't be returned to the pool.  If ever conn is unusable after use, I don't see a reason to use a pool.